.github/workflows/actions/setup_dev_ubuntu/action.yaml

```diff
@@ -0,0 +1,84 @@
+name: 'Setup dev environment'
+description: 'Setup a dev environment for rgh-action Ubuntu'
+inputs:
+  version_ghc:
+    description: 'GHC version needed'
+    required: false
+    default: 8.10.7
+  version_cabal:
+    description: 'Cabal version needed'
+    required: false
+    default: 3.10.1.0
+  cache_prefix:
+    description: 'A prefix to insert to the cache id'
+    require: false
+    default: ''
+  clean_caches:
+    description: 'Should we clean old caches for this branch and keep only the last one?'
+    require: false
+    default: true
+runs:
+  using: "composite"
+  steps:
+
+    - name: "Clear old caches"
+      if: ${{ inputs.clean_caches == 'true' }}
+      shell: bash
+      run: |
+        .github/workflows/actions/setup_dev_ubuntu/clear_caches
+      env:
+          PREFIX: ${{ inputs.cache_prefix }}
+          GH_TOKEN: ${{ github.token }}
+
+    - name: "Install packages"
+      shell: bash
+      run: |
+        .github/workflows/actions/setup_dev_ubuntu/install_packages
+
+    - name: Cache environment setup
+      id: cache_env
+      uses: actions/cache@v3
+      with:
+        path: |
+          /usr/local/.ghcup/ghc/${{ inputs.version_ghc }}
+          /usr/local/.ghcup/bin/cabal-${{ inputs.version_cabal }}
+          /usr/local/.ghcup/bin/cabal
+          /usr/local/.ghcup/bin/ghc
+          ${{ github.workspace }}/srv/
+        key: env-${{ runner.os }}-${{ hashFiles('.github/workflows/actions/setup_dev_ubuntu/install_libraries') }}
+
+    - name: "Setup build environment"
+      if: ${{ steps.cache_env.outputs.cache-hit != 'true' }}
+      shell: bash
+      run: |
+        .github/workflows/actions/setup_dev_ubuntu/install_libraries
+
+    - name: "Setup PATH"
+      shell: bash
+      run: |
+        echo "$(realpath $PWD/srv)/bin" >> $GITHUB_PATH
+
+    - name: Cache cabal storage
+      id: cache_cabal
+      uses: actions/cache@v3
+      with:
+        path: |
+          ~/.cache/cabal/packages
+          ~/.local/state/cabal
```

Comment:
Not sure what `~/.local/state/cabal` is used for, but you definitely want to cache `~/.cabal/store`.

Reply:
For some reason, this is where cabal stores its stuff on github action Ubuntu servers.

By the way, I opened this [PR](https://github.com/actions/cache/pull/1183) to fix GitHub cache doc.